internal/pkg/describe/env_test.go

```diff
@@ -152,3 +152,155 @@ func TestEnvDescriber_Describe(t *testing.T) {
 		})
 	}
 }
+
+func TestEnvDescriber_JSONString(t *testing.T) {
+	testProject := &archer.Project{
+		Name: "testProject",
+		Tags: map[string]string{"key1": "value1", "key2": "value2"},
+	}
+	testEnv := &archer.Environment{
+		Project:          "testProject",
+		Name:             "testEnv",
+		Region:           "us-west-2",
+		AccountID:        "123456789012",
+		Prod:             false,
+		RegistryURL:      "",
+		ExecutionRoleARN: "",
+		ManagerRoleARN:   "",
+	}
+	testApp1 := &archer.Application{
+		Project: "testProject",
+		Name:    "testApp1",
+		Type:    "load-balanced",
+	}
+	testApp2 := &archer.Application{
+		Project: "testProject",
+		Name:    "testApp2",
+		Type:    "load-balanced",
+	}
+	testApp3 := &archer.Application{
+		Project: "testProject",
+		Name:    "testApp3",
+		Type:    "load-balanced",
+	}
+	allApps := []*archer.Application{testApp1, testApp2, testApp3}
+	testCases := map[string]struct {
+		setupMocks func(mocks envDescriberMocks)
+
+		shouldOutputJSON bool
+		wantedError      error
+		wantedContent    string
+	}{
+		"correctly shows json output": {
+			shouldOutputJSON: true,
+			wantedContent:    "{\"environment\":{\"project\":\"testProject\",\"name\":\"testEnv\",\"region\":\"us-west-2\",\"accountID\":\"123456789012\",\"prod\":false,\"registryURL\":\"\",\"executionRoleARN\":\"\",\"managerRoleARN\":\"\"},\"applications\":[{\"project\":\"testProject\",\"name\":\"testApp1\",\"type\":\"load-balanced\"},{\"project\":\"testProject\",\"name\":\"testApp2\",\"type\":\"load-balanced\"},{\"project\":\"testProject\",\"name\":\"testApp3\",\"type\":\"load-balanced\"}],\"tags\":{\"key1\":\"value1\",\"key2\":\"value2\"}}\n",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
```

Comment:
We only have to write [table-driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) if we are testing different cases. In this situation we don't, so we can get rid of the `map` and for-loop.

Reply:
Oh, that makes sense. Looks like I don't have to test with/without resources separately--I guess that logic doesn't live in these methods. Thanks!